client/client.go

```diff
@@ -15,88 +15,85 @@
 package client
 
 import (
-	"fmt"
-	"net/http"
-	"net/url"
+	"log"
 
 	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/client/restclient"
-	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/client/unversioned"
 )
 
-type AppControllerClient struct {
-	*http.Client
-	Root                   *url.URL
-	dependenciesURL        *url.URL
-	resourceDefinitionsURL *url.URL
+type Interface interface {
```

Comment:
I think this is a pattern that is also existing in kubernetes core and we should follow it: https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/client.go#L30